docs/mathematics/happy-number.mdx

```diff
@@ -0,0 +1,244 @@
+  ]
+}>
+<TabItem value="java">
+
+```java
+import java.util.*;
+
+class HappyNumber{
+   
+   /*This function takes a number as input and returns
+     sum of its digits.*/
+   static int sumOfSquares(int n){
+      
+       int sum = 0;
+       while(n != 0){
+           /*Taking mod of number with 10 each time
+            gives its units place digit. We add its
+            square to the sum.*/
+           int d = n%10;
+           sum += d*d;
+           //Replace the number with number / 10.
+           n = Math.floorDiv(n,10);
+       }
+
+       return sum;
+   }   
+       
+   /*This function takes a number as input and checksif the number is a 
+     happy number.*/
+   static boolean happyNumber(int n){
+       
+       int first = n;
+       int second = n;
+
+       while(true){
+
+          //The first variable calling sum of digits function once 
+          first = sumOfSquares(first);
+
+          //The second variable calling sum of digits function twice
+          second = sumOfSquares(second);
+          second = sumOfSquares(second);
+
+          /*If the first one equals the second, we break out of the loop. Otherwise,
+            it will run into an infinite loop.*/
+          if(first == second)
+             break;
+          else
+             continue;    
+       }   
+
+       /*The number is happy if at the end of iterations the number
+          becomes 1. If it is one, then we return True*/ 
+       if(first == 1)
+          return true;
+       else
+          return false; 
+
+   }
+
+   public static void main(String[] args){
+     
+       Scanner scan = new Scanner(System.in);
+       System.out.print("Enter a number to check if it's a happy number or not: ");
+       //Taking input from the user 
+       int n = scan.nextInt();
+       scan.close();       
+       
+       //checker
+       boolean check = happyNumber(n);      
+        
+       //If the above function returns True, it means number is a happy number
+       if(check == true){
+          System.out.println("The given number "+n+" is a happy number!");
+       }
+       else
+          System.out.println("The given number "+n+" is not a happy number!");   
+   }
+
+}
+```
+
+</TabItem>
+<TabItem value="py">
+
+```py
+def sum_square_digits(number):
+
+    sum_squares = 0
+    
+    while(number):
+        ''' Taking mod of number with 10 each time
+            gives its units place digit. We add its
+            square to the sum.'''
+        sum_squares += (number % 10) ** 2;
+        # Replace the number with number / 10.
+        number = number // 10;
+    
+    return sum_squares
+
+''' This function takes a number as input and checks
+    if the number is a happy number. The idea behind
+    the code is - we maintain two variables. The second
+    variable is one step ahead of the first one. If
+    the first variable becomes 1 so the second will be
+    1 anyhow, because 1*1 = 1. So, we terminate the loop.'''
```

Comment:
Can you shorten this a bit to make it crisp?

Reply:
actually, this is explained in the later code part so it will be good if I shorten it.